src/fwrite.c

```diff
@@ -560,14 +560,11 @@ void writeCategString(const void *col, int64_t row, char **pch)
 
 #ifndef NOZLIB
 int init_stream(z_stream *stream) {
-  memset(stream, 0, sizeof(z_stream)); // shouldn't be needed, done as part of #4099 to be sure
   stream->next_in = Z_NULL;
   stream->zalloc = Z_NULL;
   stream->zfree = Z_NULL;
   stream->opaque = Z_NULL;
-
-  // 31 comes from : windows bits 15 | 16 gzip format
-  int err = deflateInit2(stream, Z_DEFAULT_COMPRESSION, Z_DEFLATED, 31, 8, Z_DEFAULT_STRATEGY);
+  int err = deflateInit2(stream, gzip_level, Z_DEFLATED, -15, 8, Z_DEFAULT_STRATEGY);
```

Comment:
why the change 31 --> -15? `-15` looks pretty magical to me, we should add a comment on how to think of it.

Reply:
OK. Add a comment in 117ab456